### PR TITLE
feat(model): add search video

### PR DESCRIPTION
**Pull request by dp0973**

### CODE REVIEW

File: holodex/http.py [hunk 1]
```diff
@@ -56,3 +56,6 @@ async def get_videos_from_channel(
         return await self.request(
             "GET", f"/channels/{channel_id}/{type}", params=params
         )
+
+    async def get_search_video(self, params: dict[str, Any]) -> Any:
+        return await self.request("POST", "search/videoSearch", params=params)
```
Comment (ombe1229) [inline holodex/http.py:61]:
```suggestion
        return await self.request("POST", "/search/videoSearch", params=params)
```

File: holodex/model/search_video.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+class SearchVideo:
+    def __init__(self, response: Any) -> None:
+        self._response = response
+
+    @property
+    def total(self) -> int:
+        return self._response["total"]
+
+    @property
+    def contents(self) -> list[SearchVideoInfo]:
```
Comment (ombe1229) [inline holodex/model/search_video.py:88]:
```suggestion
    def items(self) -> list[SearchVideoInfo]:
```

File: holodex/client.py [hunk 3]
```diff
@@ -70,3 +71,19 @@ async def videos_from_channel(
...
+        offset: int,
+        limit: int,
+        sort: Literal["oldest", "newest"] = "newest",
+        lang: Optional[Literal["all", "en", "ja"]] = None,
+        target: Optional[Literal["clip", "stream"]] = None,
+        conditions: Optional[list[Condition]] = None,
+        topic: Optional[list[str]] = None,
+        vch: Optional[list[str]] = None,
+        org: Optional[list[str]] = None,
+        paginated: bool = False,
```
Comment (ombe1229) [inline holodex/client.py]:
```suggestion
        paginated: Optional[bool] = None,
```

File: holodex/client.py [hunk 4]
```diff
@@ -70,3 +71,19 @@ async def videos_from_channel(
...
+    ) -> SearchVideo:
+        params = self.__get_params(locals())
```
Comment (ombe1229) [inline holodex/client.py:87]:
```suggestion
        params = self.__get_params(locals(), ["offset", "limit", "sort"])
```

**Comment by ombe1229** [comment]
format

**Comment by dp0973** [comment]
Eng e

**Review by ombe1229 — APPROVED** [review]

**Pull request merged**